src/simsopt/util/dipole_array_helper_functions.py

```diff
@@ -0,0 +1,754 @@
+            base_wp_curves.append(curve)
+    return base_wp_curves
+
+
+def generate_tf_array(winding_surface, ntf, TF_R0, TF_a, TF_b, fixed_geo_tfs=False, planar_tfs=True, order=6, numquadpoints=32):
+    """
+    Initialize an array of planar toroidal field coils over a half field period
+    Parameters:
+        winding_surface: surface upon which to obtain field periodicity/stellarator symmetry for the coils
+        ntf: number of TF coils per half field period
+        TF_R0: major radius of the TF coils
+        TF_a: minor radius of the TF coils (in R direction)
+        TF_b: minor radius of the TF coils (in Z direction)
+        fixed_geo_tfs: whether to fix the geometric degrees of freedom of the TF coils
+        numquadpoints: number of quadrature points representing each coil
+    Returns:
+        base_tf_curves: list of initialized curves (half field period)
+    """
+    from simsopt.geo import create_equally_spaced_curves
+    if not fixed_geo_tfs:
+        if planar_tfs:
+            try:
+                from simsopt.geo import create_equally_spaced_cylindrical_curves
+                base_tf_curves = create_equally_spaced_cylindrical_curves(ntf, winding_surface.nfp, stellsym=winding_surface.stellsym, R0=TF_R0, a=TF_a, b=TF_b, numquadpoints=numquadpoints)
+            except ImportError:
+                raise ImportError("Need to be on the windowpane branch with the correct TF curve class to unfix TF geometry")
+        else:
+            base_tf_curves = create_equally_spaced_curves(ncurves=ntf, nfp=winding_surface.nfp, stellsym=winding_surface.stellsym, R0=TF_R0, R1=TF_a, order=order, numquadpoints=numquadpoints)
+    else:
+        # order=1 is fine for elliptical
+        base_tf_curves = create_equally_spaced_curves(ncurves=ntf, nfp=winding_surface.nfp, stellsym=winding_surface.stellsym, R0=TF_R0, R1=TF_a, order=order, numquadpoints=numquadpoints)
+        # add this for elliptical TF coils - keep same ellipticity as VV
+        for c in base_tf_curves:
+            c.set("zs(1)", -TF_b)  # see create_equally_spaced_curves doc for minus sign info
+
+    return base_tf_curves
+
+
+def generate_curves(surf, VV, planar_tfs=False, outdir='',
+                    inboard_radius=0.8, wp_fil_spacing=0.75, half_per_spacing=0.75, wp_n=2,
+                    numquadpoints=32, order=12, verbose=True,
+                    fixed_geo_tfs=False, tf_init_fac=2,
+                    ntf=3,
+                    ):
+    """
+    Generate the curves for the winding surface and TF coils.
+
+    Args:
+        VV: SurfaceRZFourier object
+            The winding surface to put the dipole coils on.
+        surf: SurfaceRZFourier object   
+            The plasma boundary surface.
+        planar_tfs: bool
+            Whether to use planar TF coils.
+        outdir: str
+            The output directory for the generated curves.
+    """
+    from simsopt.geo import curves_to_vtk
```

Comment:
a few of these functions import `curves_to_vtk` could make the code cleaner by making it a global import like `numpy` is